test/e2e/helpers.go

```diff
@@ -946,3 +936,9 @@ func CopyConfigMap(ctx context.Context, input clusterctl.ApplyCustomClusterTempl
 		}
 	}, input.WaitForControlPlaneIntervals...).Should(Succeed())
 }
+
+func getSubscriptionID() string {
+	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
+	Expect(subscriptionID).NotTo(BeEmpty())
```

Comment:
I don't think it's an issue, but if this were ever called in a context using a non-default `Gomega` handle then this might misbehave, like in an `Eventually` invoking a `func(g Gomega)`. It might be safer but more verbose then to pass a `Gomega` here as a parameter:
```suggestion
func getSubscriptionID(g Gomega) string {
	subscriptionID := os.Getenv("AZURE_SUBSCRIPTION_ID")
	g.Expect(subscriptionID).NotTo(BeEmpty())
```

I think the default used by plain `Expect` is called `Default`.

Reply:
>  this might misbehave, like in an `Eventually` invoking a `func(g Gomega)`

As in `CopyConfigMap`, the function ten lines above this one. 😄

I agree that's safer–I'll make that change.